src/components/utils/hooks/useGetMessages.tsx

```diff
@@ -0,0 +1,27 @@
+import { Conversation } from '@xmtp/xmtp-js';
+import { useEffect } from 'react';
+import { useMessageStore } from 'src/store/message';
+
+const useGetMessages = (conversation?: Conversation) => {
+  const messageState = useMessageStore((state) => state);
+  const { messages, setMessages } = messageState;
+
+  useEffect(() => {
+    if (!conversation) {
+      return;
+    }
+    const loadMessages = async () => {
+      const newMessages = await conversation.messages();
```

Comment:
Don't worry @bigint, we are going to add pagination to make this more performant. Just prioritizing getting the UI in place first.

Reply:
Cool ⚡